src/com/magento/idea/magento2plugin/inspections/xml/ObserverDeclarationInspection.java

```diff
@@ -0,0 +1,247 @@
+package com.magento.idea.magento2plugin.inspections.xml;
+
+import com.intellij.codeInspection.ProblemHighlightType;
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.ide.highlighter.XmlFileType;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.*;
+import com.intellij.psi.search.GlobalSearchScope;
+import com.intellij.psi.util.PsiTreeUtil;
+import com.intellij.psi.xml.XmlAttribute;
+import com.intellij.psi.xml.XmlAttributeValue;
+import com.intellij.psi.xml.XmlDocument;
+import com.intellij.psi.xml.XmlTag;
+import com.jetbrains.php.lang.inspections.PhpInspection;
+import com.magento.idea.magento2plugin.indexes.EventIndex;
+import com.magento.idea.magento2plugin.magento.files.ModuleXml;
+import com.magento.idea.magento2plugin.magento.packages.MagentoPackages;
+import org.jetbrains.annotations.NotNull;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import com.intellij.openapi.vfs.VfsUtil;
+import org.jetbrains.annotations.Nullable;
+
+public class ObserverDeclarationInspection extends PhpInspection {
+
+    @NotNull
+    @Override
+    public PsiElementVisitor buildVisitor(@NotNull ProblemsHolder problemsHolder, boolean b) {
+        return new XmlElementVisitor() {
+            private final String moduleXmlFileName = ModuleXml.getInstance().getFileName();
+            private static final String eventsXmlFileName = "events.xml";//todo: move to separate file as done to ModuleXml
+            private static final String duplicatedObserverNameSameFileProblemDescription = "Observer name already used in this file. For more details see Inspection Description.";
```

Comment:
The Observer name already used in this file. For more details see Inspection Description.